mbq/metrics/utils.py

```diff
@@ -1,8 +1,14 @@
-def tag_dict_to_list(tags):
-    return sorted([
-        ':'.join([str(k), str(v)]) if v else k
-        for k, v in tags.items()
-    ])
+def tags_as_list(tags):
+    if tags is None:
+        return []
+    if isinstance(tags, (list, tuple)):
+        return list(tags)
+    elif isinstance(tags, dict):
+        return sorted([
+            ':'.join([str(k), str(v)]) if v else k
+            for k, v in tags.items()
+        ])
+    raise ValueError('unexpected type for parameter tags')
```

Comment:
Reworked this function. Idea is to immediately drop down to lists everywhere in the Collector whenever we take in `tags` from the user. Addition of defaulting inside this function eliminates the default handling from each function in the Collector.